lib/reply.js

```diff
@@ -406,11 +438,25 @@ function onSendEnd (reply, payload) {
     }
 
     res.writeHead(statusCode, reply[kReplyHeaders])
+
     // avoid ArgumentsAdaptorTrampoline from V8
     res.end(null, null, null)
     return
   }
 
+  // it is intended to not sending trailers when payload is not set
+  // the reason behind is when there is no payload
+  // there is no meanings for a dynamic trailers
+  // we check if we need to update the trailers header and set it
+  const trailerHeaders = Object.keys(reply[kReplyTrailers])
+  const isTEAllowed = req.headers && typeof req.headers.te === 'string' && req.headers.te.includes('trailers')
+  reply[kReplyShouldSendTrailers] = trailerHeaders.length > 0 && isTEAllowed && payload !== undefined && payload !== null && payload !== ''
```

Comment:
It should be `payload !== undefined && payload !== null`, but I am not sure if it is bug or somthing else. It will be serialized and return stringify `undefined` and `null`. 

So, I added `payload !== ''` in the end.

Reply:
`JSON. stringify` has been used on `payload` when you use this judgement. So object `null` has been translated to string `'null'`. When you use `reply.send(undefined)` or `reply.send('')`, it will mark payload as empty. But when you use `reply.send(null)`, it will response string `'null'`. It's a feature that's always been there. So I think is ok to use `&& payload` here.